<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="database_path"     default="rtabmap_robot_2.db"/>
  <!-- <arg name="database_path"     default="rtabmap.db"/> -->
  <arg name="args"              default=""/>
  
  <arg name="wait_for_transform"  default="0.2"/>

  <group ns="/robot_2">
    <include file="$(find riltaur_rrt_exploration)/launch/includes/move_base_multiple_robots.launch">
      <arg name="namespace" value="robot_2"/>
      <!--<arg name="database_path"     default="rtabmap.db"/>
      <arg name="args"              default=""/>
      <arg name="wait_for_transform"  default="0.2"/> -->
    </include> 
  </group>

  <group ns="rtabmap_2">
    <node name="rtabmap_2" pkg="rtabmap_ros" type="rtabmap" output="screen" args="$(arg args)">
      <param name="database_path"       type="string" value="$(arg database_path)"/>
      <param name="frame_id"            type="string" value="robot_2/base_footprint"/>
      <param name="odom_frame_id"       type="string" value="robot_2/odom"/>
      <param name="wait_for_transform_duration"  type="double"   value="$(arg wait_for_transform)"/>
      <param name="subscribe_depth"     type="bool"   value="true"/>
      <param name="subscribe_scan"      type="bool"   value="true"/>

      <param name="map_frame_id"         type="string" value="robot_2/map"/>
      <!--<param name="queue_size"           type="int" value="100"/> 
      <param name="publish_tf"           type="bool"   value="false"/>
      
      <param name="landmark_linear_variance"   type="double" value="0.0001"/>
      <param name="landmark_angular_variance"  type="double" value="9999"/> -->

    
      <!-- inputs -->
      <remap from="scan"            to="/robot_2/scan"/>
      <remap from="rgb/image"       to="/robot_2/camera/color/image_raw"/>
      <remap from="depth/image"     to="/robot_2/camera/depth/image_raw"/>
      <remap from="rgb/camera_info" to="/robot_2/camera/color/camera_info"/>  
      <remap from="odom"            to="/robot_2/odom"/>         
      
      <!-- output -->
      <remap from="grid_map" to="/robot_2/map"/> 
    
      <!-- RTAB-Map's parameters: do "rosrun rtabmap rtabmap (double-dash)params" to see the list of available parameters. -->
      <param name="RGBD/ProximityBySpace"        type="string" value="true"/>   
      <param name="RGBD/OptimizeFromGraphEnd"    type="string" value="false"/>  
      <param name="Kp/MaxDepth"                  type="string" value="4.0"/>
      <param name="Reg/Strategy"                 type="string" value="1.0"/>       <!-- #1 >-->      
      <param name="Icp/CoprrespondenceRatio"     type="string" value="0.15"/>
      <param name="Vis/MinInliers"               type="string" value="50"/>        <!-- #5.0 >-->
      <param name="Vis/InlierDistance"           type="string" value="0.1"/>    
      <param name="RGBD/AngularUpdate"           type="string" value="0.1"/>    
      <param name="RGBD/LinearUpdate"            type="string" value="0.1"/>    
      <param name="Rtabmap/TimeThr"              type="string" value="0"/>
      <param name="Mem/RehearsalSimilarity"      type="string" value="0.30"/>
      <!-- <param name="Optimizer/Slam2D"             type="string" value="true"/> -->
      <param name="Reg/Force3DoF"                type="string" value="true"/>
      <param name="Grid/FromDepth"               type="string" value="false"/>
      <param name="RGBD/ProximityPathMaxNeighbors" type="string" value="10"/>


      <!-- param name="Grid/FromDepth"            type="string" value="false"/ 
       <param name="RGBD/ProximityPathMaxNeighbors" type="string" value="10"/>   
      <param name="Odom/Strategy"              value="1"/>
      <param name="Vis/Cortype"                value="1"/>
      <param name="GFTT/MinDistance"             type="string" value="10"/>
      <param name="grid_size"                    type="double" value="50"/>
      <param name="cloud_noise_filtering_radius" value="0.05"/>
      <param name="cloud_noise_filtering_min_neighbors" value="2"/>
      <param name="proj_max_ground_angle" value="45"/>
      <param name="proj_max_ground_height" value="0.1"/>
      <param name="proj_max_height" value="2.0"/>
      <param name="proj_min_cluster_size" value="20"/> -->


      <!-- param name="Mem/IncrementalMemory"        type="string" value="true"/>  -->

    </node>
  </group>
</launch>